crates/biome_json_syntax/src/file_source.rs

```diff
@@ -1,45 +1,32 @@
 use biome_rowan::FileSourceError;
 use std::path::Path;
 
-// TODO: Jsonc is not well-defined, so some files may only support comments
-// but no trailing commas. We should reconsider whether to use Jsonc as a variant,
-// or just use something like: JsonWithComments, JsonWithCommentsAndTrailingCommas
-//
-// Currently, the "variant" key and other properties are making things duplicated
-// and error-prone.
-
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(
     Debug, Clone, Default, Copy, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize,
 )]
 pub struct JsonFileSource {
-    variant: JsonVariant,
     allow_trailing_commas: bool,
     allow_comments: bool,
 }
 
-#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
-#[derive(
-    Debug, Clone, Default, Copy, Eq, PartialEq, Hash, serde::Serialize, serde::Deserialize,
-)]
-enum JsonVariant {
-    #[default]
-    Standard,
-    Jsonc,
-}
-
 impl JsonFileSource {
     pub fn json() -> Self {
         Self {
-            variant: JsonVariant::Standard,
             allow_trailing_commas: false,
             allow_comments: false,
         }
     }
 
-    pub fn jsonc() -> Self {
+    pub fn json_with_comments() -> Self {
```

Comment:
I think a more flexible way to define the file source is to provide builder pattern methods at this point. We can keep `json()`, and provide the following methods too:

```rs
impl JsonFileSource {
	fn with_comments(mut self) -> Self {}
	fn with_trailing_commas(mut self) -> Self {}
}
```

This is a pattern we use in the whole code base

Reply:
We already have that. I can use them.

Reply:
That would be great, it allows for more flexibility